### PR TITLE
Full function building at compile time and test distributed

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

File: src/dsl/model_macro.jl [hunk 1]
```diff
@@ -289,42 +289,92 @@ end
...
-      ___D = Differential(t)
-      $params
-      $bvars
+      $fname = $f_ex
       $diffeq
     end
   end
 end
+
+function convert_rhs_to_Expression(s::Symbol, dvars, params, t)
```
Comment (ChrisRackauckas) [inline src/dsl/model_macro.jl]: It might be good to upstream this to MT.

**Comment by ChrisRackauckas** [comment]
@HarrisonGrodin @YingboMa take note that this is what I meant by compile-time usage. 

@andreasnoack @vjd this should fix the distributed issues, and a test was added for that.

File: src/dsl/model_macro.jl [hunk 2]
```diff
@@ -289,42 +289,92 @@ end
...
+  j = findfirst(params) do x
+    if x isa Operation
+      x.op.name == ex.args[1]
+    else # For CL(t)
+      x.name == ex.args[1]
+    end
+  end
+
+  if i === j === nothing
+    op = eval(ex.args[1])  # HACK
```
Comment (andreasnoack) [inline src/dsl/model_macro.jl]: Any chance that we can avoid more `eval`ing in macros? If not now then maybe later?

Comment (ChrisRackauckas) [reply src/dsl/model_macro.jl]: what's your suggestion for how to handle this?

Comment (ChrisRackauckas) [reply src/dsl/model_macro.jl]: found a way.

File: test/core/dsl.jl [hunk 3]
```diff
@@ -38,6 +38,7 @@ mdsl = @model begin
 
     @vars begin
       conc = Central / V
+      conc2 = Central^2
```
Comment (ChrisRackauckas) [inline test/core/dsl.jl]: made this test a little harder.

**Comment by ChrisRackauckas** [comment]
It was just a subtle issue with processes in a module, so a general Julia issue. Once that was avoided, tests worked fine. The result is no evals and distributed works in all cases.

**Comment by pkofod** [comment]
Strange, but great you figured it out. Was there already an issue for this?

**Comment by ChrisRackauckas** [comment]
Not sure if there's an issue, but it's really just because `@everywhere` and such hard code their module to be Main

**Pull request merged**